### ISSUE TITLE
Consider .local as a valid top level domain name

### ISSUE THREAD

**Issue by zdenda** [mozilla-mobile/android-components#4757]
I'm redirected to search results when I try to go to our company's intranet at "intranet.local" in the new Firefox Preview.
I have to fill in the whole URL with scheme (http://intranet.local/) to get there.

**Steps to reproduce:**
1. Open Fenix / Firefox Preview (version: Nightly 191010)
2. Enter `intranet.local` into the address bar
3. Hit the Go button on keyboard

**Expected behavior:**
Web page at http://intranet.local/ is opened (or if it doesn't exist "address not found" is shown)

**Actual behavior:**
Search results for "intranet.local" are opened

I looked at the source code, and I think I can send a PR with fix, if you think that this is a valid issue.
Maybe even add other special top level DNS names from [rfc2606](https://tools.ietf.org/html/rfc2606#section-2).

**Comment by Amejia481** [comment]
I was testing with the latest Fenix nightly [1] and it was behaving as expected. As I'm not in network with a `intranet.local`, I'm getting a "Address Not Found" error page. Could you please test if it is working for you? 

```
[1]
Nightly 200107 06:01 (Build #20070607)
27.0.0, 707874f68
GV: 73.0a1-20200106092427
Tuesday 1/7 @ 6:06 AM
```

**Comment by zdenda** [comment]
Yes, I can confirm that it's working in the latest nightly. I guess it was fixed by 4af24d9b71ae952dc8e07e15a1e5db5b524fabc5.
